pkg/serverless/metrics/enhanced_metrics.go

```diff
@@ -188,14 +188,21 @@ func GenerateEnhancedMetricsFromReportLog(args GenerateEnhancedMetricsFromReport
 		SampleRate: 1,
 		Timestamp:  timestamp,
 	})
-	args.Demux.AggregateSample(metrics.MetricSample{
-		Name:       postRuntimeDurationMetric,
-		Value:      postRuntimeDuration,
-		Mtype:      metrics.DistributionType,
-		Tags:       args.Tags,
-		SampleRate: 1,
-		Timestamp:  timestamp,
-	})
+	if args.RuntimeStart.IsZero() || args.RuntimeEnd.IsZero() {
```

Comment:
I wonder if we could keep a running tail of all of the requests instead of defaulting to not calculating it. For instance, whenever we get the REPORT/END log lines, we could add them to a global map, and then calculate them at that point instead.

Reply:
I think I get what you mean. So we map `reqId` -> `start|runtimeDone` logs, and only when we have both items then we calculate the runtime duration metrics

Reply:
Yes, exactly. Does that sound like a good idea to you? There's probably some considerations (i.e, we may want to expire requests after a while if we don't receive a runtimeDone log) but besides that I think it's fairly reasonable.

Reply:
👍 yup that makes perfect sense. I'll log a card for us